specifications/KeyValueStore/ClientCentric.tla

```diff
@@ -0,0 +1,185 @@
+--------------------------- MODULE ClientCentric ---------------------------
+EXTENDS Naturals, TLC, Sequences, FiniteSets, Util
+VARIABLES Keys, Values
```

Comment:
ClientCentric is not my module, written by Tim, so I am reluctant to make any change there.

Reply:
/cc @TimSoethout

Reply:
The module is instantiated as seen in: https://github.com/tlaplus/Examples/pull/84/files/adb3d2aa3f62812a2becd5c2f6700d69eb1c5134#diff-ccf178cee4a4265b5f8405a78d596e79973a841e20119a281c19775a1ecae139R14
`CC == INSTANCE ClientCentric WITH Keys <- Key, Values <- TxId \union {NoVal} `

If something similar is possible with constant parameters, and it is more idiomatic TLA, you should definitely go for parameters.

Reply:
Instantiation is certainly possible for constant as well as variable parameters. Declaring Keys and Values to be variables allows these sets to change during an execution, which might be interesting for some applications, but I don't know what the original paper had in mind. If they are constants, the module is a constant module (since all operator definitions are of constant level as well), which allows operators to be instantiated at constant, variable or action level. Potential complications with non-constant modules are described in section 17.8 of Specifying Systems. However, my comment was just a caveat for later, there is nothing wrong with the modules as they are written, and they are accepted by TLC.